src/core/Akka.Tests/Actor/RepointableActorRefSpecs.cs

```diff
@@ -0,0 +1,57 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="RepointableActorRefSpecs.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.TestKit;
+using Xunit;
+
+namespace Akka.Tests.Actor
+{
+    public class RepointableActorRefSpecs : AkkaSpec
+    {
+        public class Bug2182Actor : ReceiveActor, IWithUnboundedStash
```

Comment:
Simplified @kantora's reproduction scenario and was able to reproduce the error using this spec.